### DISCUSSION TITLE
Is it possible to convert a repository to a branch?

### DISCUSSION THREAD

**Discussion by benyltam** [community/community#33549] [Actions]
Is it possible to convert a repository to a branch?

**Comment by radar-blip** [comment]
What do you mean? Your repository is already the 'main' branch.
You want to create an additional branch?

**Reply by benyltam** [reply]
What I want to do are:
1. Convert the current repository to a branch.
2. Remove the content of the current repository.
3. Download a solution from my Visual Studio to the current repository for the MSDN experts to debug.

Can you help me?

**Reply by airtower-luna** [reply]
That doesn't make sense: A branch is a *part* of a repository. Within the repository you can create and delete branches as you like.

**Reply by benyltam** [reply]
Hi airtower,

Thanks for your reply. Is it impossible to carry out what I want to do? Or, is there any similar way to implement those I want to do?

**Reply by airtower-luna** [reply]
No, what I'm saying is that the question doesn't make sense, at least the way it's phrased. It's like "can I make a motor a car?" If you can be more precise about what you need maybe someone can help, or suggest an alternative.

**Comment by jfullstackdev** [comment]
just want to comment on too, if I'm getting what you are trying to do,

convert a current repository to a branch, remove the content of the `main` branch, download a solution from VS for others to debug, I'm expecting because you don't want to change the content of the main branch ? 

or is it required for that solution from VS Code to run it not on the `main` branch ?

because branches are not required to be merged all the time, so others can check that without affecting the `main` branch 

that might be your problem as you think it will affect the `main` branch entirely

**Comment by JaCoB1123** [comment]
Since this showed up on Google, here's how I merged two repositories into one.

Let's say I have two repositories `main` and `other` both with a branch main:

```
cd main
```

Add other repository as a remote and fetch it's `main` branch, creating the ref `other/main`

```
git remote add other ../other
git fetch other main
```

Create new branch called `other` for the fetched code and optionally push new branch to the regular remote.

```
git checkout other/main -b other
git push -u origin other
```

Now you have branch main with your original code and branch other containing the code from the others' main branch. If that's the only branch in the other repository, you could delete it now.
